### PR TITLE
Fix timeout reporting in TAP reporter

### LINKED ISSUE #2612: TAP output does not say which tests (or at least which file) failed after timeout
So here is the output of timedout test report (there are passed tests as well).
```
# Exited because no new tests completed within the last 180000ms of inactivity
not ok 51 - Exited because no new tests completed within the last 180000ms of inactivity
```
It is impossible to figure out which test caused timeout. 
Would be nice at least to narrow it down to test file, to understand where to start searching.

### PR DESCRIPTION
**Pull request by tymfear**
Fixes #2612

### CODE REVIEW

**Comment by tymfear** [comment]
Example of output
```
not ok 2 - test23.ts › Exited because no new tests completed within the last 9000ms of inactivity
not ok 3 - dir › test.ts › Exited because no new tests completed within the last 9000ms of inactivity
```

**Comment by novemberborn** [comment]
I'm struggling to understand why this needs a change in `api.js`. I had assumed the TAP reporter was lagging behind the default reporter, but this implies it's now doing something the default reporter cannot? In other words, why aren't the existing status events sufficient?

**Comment by tymfear** [comment]
@novemberborn yeah, you're right. Have found the the `pendingTests` set available for event. Will make an update.

**Comment by tymfear** [comment]
@novemberborn, so how about new approach - report each pending test as failed when timeout occurs?

File: lib/reporters/tap.js [hunk 1]
```diff
@@ -131,6 +131,16 @@ class TapReporter {
 		}
 	}
 
+	writeTimeout(evt) {
+		const err = new Error(`Exited because no new tests completed within the last ${evt.period}ms of inactivity`);
+
+		for (const [testFile, tests] of evt.pendingTests) {
+			[...tests].forEach(title => {
+				this.writeTest({testFile, title, err}, {passed: false, todo: false, skip: false});
+			});
```
Comment (novemberborn) [inline lib/reporters/tap.js:140]:
```suggestion
			for (const title of tests) {
				this.writeTest({testFile, title, err}, {passed: false, todo: false, skip: false});
			}
```

File: lib/reporters/tap.js [hunk 2]
```diff
@@ -209,4 +219,5 @@ class TapReporter {
 		}
 	}
 }
+
```
Comment (novemberborn) [inline lib/reporters/tap.js:222]:
```suggestion
```

**Review by novemberborn — COMMENTED** [review]
@tymfear yes that's much better, thank you. What does this look like in the output?

I can't recall if we have timeout integration tests for the other reporters. If we do, could you add the same for the TAP reporter?

File: lib/reporters/tap.js (hunk 1 above)
Comment (tymfear) [reply lib/reporters/tap.js]: Done

**Comment by tymfear** [comment]
@novemberborn  - I've updated my comment with output. Probably should've rather add a new one, sorry.
I couldn't find any reporter tests with timeouts.

**Review by novemberborn — APPROVED** [review]

**Comment by novemberborn** [comment]
Thanks @tymfear and happy new year 🎆

**Pull request merged**